app/components/steps/why.js

```diff
@@ -0,0 +1,254 @@
+//https://github.com/EnCiv/civil-pursuit/issues/103
+//https://github.com/EnCiv/civil-pursuit/issues/214
+
+import React, { useEffect, useContext, useRef, useState } from 'react'
+import DeliberationContext from '../deliberation-context'
+import WhyInput from '../why-input'
+import { H, Level } from 'react-accessible-headings'
+import cx from 'classnames'
+import { createUseStyles } from 'react-jss'
+import { isEqual, cloneDeep } from 'lodash'
+
+export default function WhyStep(props) {
+  const { data, upsert } = useContext(DeliberationContext)
+  const { category, intro, ...otherProps } = props
+  const derivedPointWhyList = derivePointWhyListByCategory(data, category)
+
+  useEffect(() => {
+    if (!data?.myWhyByParentId && data?.reducedPointList?.length > 0) {
+      const ids = data.reducedPointList.map(point => point._id)
+      window.socket.emit('getUserWhys', ids, results => {
+        const myWhyByParentId = results.reduce((acc, point) => {
+          acc[point.parentId] = point
+          return acc
+        }, {})
+        upsert({ myWhyByParentId })
+      })
+    }
+  }, [data, upsert])
+
+  function handleOnDone({ valid, value, delta }) {
+    if (!delta.category) {
+      delta.category = category
+    }
+
+    const updatedMyWhyByParentId = {
+      ...data.myWhyByParentId,
+      [delta.parentId]: delta,
+    }
+
+    upsert({ myWhyByParentId: updatedMyWhyByParentId })
+
+    window.socket.emit('upsertWhy', delta, updatedDoc => {
+      if (updatedDoc) {
```

Comment:
Good point about handling the return document, but can you use isEqual to campare the returned value and not upsert if equal. This avoids unnecessary retenders. And also like above, just upsert the changed part.